src/mscorlib/src/System/Environment.Unix.cs

```diff
@@ -0,0 +1,150 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static readonly unsafe Lazy<Dictionary<string, string>> s_environ = new Lazy<Dictionary<string, string>>(() =>
+        {
+            var results = new Dictionary<string, string>();
```

Comment:
The CoreCLR PAL does the same caching. Now that this code is in CoreCLR and it does not appear that it can be shared 1:1 with CoreRT, it seems wasteful to do the caching again in C# - the environment is sizable chunk of memory.